ingenialink/dictionary.py

```diff
@@ -1,23 +1,84 @@
 from abc import ABC, abstractmethod
+import xml.etree.ElementTree as ET
+from ingenialink.register import REG_DTYPE, REG_ACCESS
+from ingenialink import exceptions as exc
 
 
-class Categories(ABC):
-    """Categories Abstract Base Class.
+class DictionaryCategories:
+    """Contains all categories from a Dictionary.
 
     Args:
-        dictionary (Dictionary): Ingenia dictionary instance.
+        dict_ (str): Path to the Ingenia dictionary.
 
     """
-    def __init__(self, dictionary):
-        self._dictionary = dictionary
 
-    @abstractmethod
-    def labels(self, category_id):
-        raise NotImplementedError
+    def __init__(self, dict_):
+        self._dict = dict_
+        self._cat_ids = []
+        self._categories = {}  # { cat_id : label }
+
+        self.load_cat_ids()
+
+    def load_cat_ids(self):
+        """Load category IDs from dictionary."""
+        with open(self._dict, 'r', encoding='utf-8') as xml_file:
```

Comment:
Avoid opening and reading the file each time you want to get some data as categories and errors. This class can get as a parameter a list of elements directly.
Same with DictionaryErrors.